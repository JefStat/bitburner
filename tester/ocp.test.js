const ocp = require('../ocp');
const { performScheduling, getRatesAtHackLevel, getNetworkStats } = require('../ocp');
const { noodles, player } = require('./StaticValues');

const mockNs = {
  getServer: () => noodles,
  getServerRequiredHackingLevel: () => noodles.requiredHackingSkill,
  getServerMoneyAvailable: () => noodles.moneyAvailable,
  getServerSecurityLevel: () => noodles.hackDifficulty,
  hasRootAccess: () => noodles.hasAdminRights,
  getServerUsedRam: () => noodles.ramUsed,
  getWeakenTime: () => 1,
  getHackTime: () => 1,
  getGrowTime: () => 1,
  formulas: {
    hacking: {
      weakenTime: () => 8000,
      growTime: () => 7000,
      hackTime: () => 1000,
      hackPercent: () => 0.001,
      growPercent: () => 0.0001,
      hackChance: () => 1,
      hackExp: () => 0.000001
    }
  }
};

describe('ocp', () => {
  beforeAll(() => {
    ocp.fakeInit(mockNs, [], player);
  });

  test('getRatesAtHackLevel', () => {
    expect(ocp.getRatesAtHackLevel(mockNs, noodles, player, 1, 64)).toMatchInlineSnapshot(`
Array [
  775.0263946437985,
  775.0263946437985,
  4.601769911504425e-7,
]
`);
  });

  test('buildServerObject', () => {
    expect(ocp.buildServerObject(mockNs, 'n00dles')).toMatchSnapshot();
  });

  test('getNetworkStats', () => {
    expect(ocp.getNetworkStats()).toMatchInlineSnapshot(`
Object {
  "listOfServersFreeRam": Array [],
  "totalFreeRam": 0,
  "totalMaxRam": 0,
  "totalUsedRam": 0,
}
`);
  });

  test('getPerformanceSnapshot', () => {
    expect(ocp.getPerformanceSnapshot(ocp.buildServerObject(mockNs, 'n00dles'), ocp.getNetworkStats())).toMatchInlineSnapshot(`
Object {
  "canBeScheduled": false,
  "maxCompleteCycles": 1,
  "optimalPacedCycles": 1,
  "percentageToSteal": 0,
}
`);
  });

  test('optimizePerformanceMetrics', () => {
    expect(ocp.optimizePerformanceMetrics(ocp.buildServerObject(mockNs, 'n00dles'))).toMatchInlineSnapshot();
  });

  test('performScheduling', () => {
    expect(ocp.performScheduling()).toMatchInlineSnapshot();
  });
});
