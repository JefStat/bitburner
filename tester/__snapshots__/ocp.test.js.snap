// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ocp buildServerObject 1`] = `
Object {
  "actualPercentageToSteal": [Function],
  "actualServerGrowthRate": [Function],
  "adjustedGrowthRate": [Function],
  "canHack": [Function],
  "cyclesNeededForGrowthCoefficient": [Function],
  "cyclesNeededForGrowthCoefficientAfterTheft": [Function],
  "getGrowThreadsNeeded": [Function],
  "getGrowThreadsNeededAfterTheft": [Function],
  "getHackThreadsNeeded": [Function],
  "getMoney": [Function],
  "getMoneyPerRamSecond": [Function],
  "getSecurity": [Function],
  "getWeakenThreadsNeededAfterGrowth": [Function],
  "getWeakenThreadsNeededAfterTheft": [Function],
  "growDelay": [Function],
  "hackDelay": [Function],
  "hasRoot": [Function],
  "isPrepped": [Function],
  "maxRam": 4,
  "minDifficulty": 1,
  "moneyMax": 1750000,
  "name": "n00dles",
  "ns": Object {
    "formulas": Object {
      "hacking": Object {
        "growPercent": [Function],
        "growTime": [Function],
        "hackChance": [Function],
        "hackExp": [Function],
        "hackPercent": [Function],
        "hackTime": [Function],
        "weakenTime": [Function],
      },
    },
    "getGrowTime": [Function],
    "getHackTime": [Function],
    "getServer": [Function],
    "getServerMoneyAvailable": [Function],
    "getServerRequiredHackingLevel": [Function],
    "getServerSecurityLevel": [Function],
    "getServerUsedRam": [Function],
    "getWeakenTime": [Function],
    "hasRootAccess": [Function],
  },
  "percentageStolenPerHackThread": [Function],
  "percentageToSteal": 0.0625,
  "prepRegressions": 0,
  "previousCycle": null,
  "previouslyPrepped": false,
  "ramAvailable": [Function],
  "serverGrowthPercentage": [Function],
  "shouldHack": [Function],
  "targetGrowthCoefficient": [Function],
  "targetGrowthCoefficientAfterTheft": [Function],
  "timeToGrow": 1,
  "timeToHack": 1,
  "timeToWeaken": [Function],
  "usedRam": [Function],
  "weakenThreadsNeeded": [Function],
}
`;
